pypdf/_page.py

```diff
@@ -576,17 +575,41 @@ def _merge_resources(
     ) -> Tuple[Dict[str, Any], Dict[str, Any]]:
         new_res = DictionaryObject()
         new_res.update(res1.get(resource, DictionaryObject()).get_object())
+
+        def compute_unique_key(base_key: str) -> Tuple[str, bool]:
+            """Find a key that either doesn't already exist or has the same
+            value (indicated by the bool)"""
+            value = page2res.raw_get(base_key)
+            # try the current key first (e.g. "foo"), but otherwise iterate
+            # through "foo-0", "foo-1", etc. new_res can contain only finitely
+            # many keys, thus this'll eventually end, even if it's been crafted
+            # to be maximally annoying.
+            computed_key = base_key
+            idx = 0
+            while computed_key in new_res:
+                if new_res.raw_get(computed_key) == value:
+                    # there's already a resource of this name, with the exact
+                    # same value
+                    return computed_key, True
+                computed_key = "%s-%d" % (key, idx)
```

Comment:
Use [f-strings](https://realpython.com/python-f-strings/) as they're much faster than [other string formatting methods.](https://www.scivision.dev/python-f-string-speed/).

```suggestion
                computed_key = f"{key}-{idx}"
```

Reply:
Ah, sorry, I thought they weren't supported in some versions of Python supported by pypdf. I take it pypdf only supports 3.6+?

Reply:
Yup:

https://github.com/py-pdf/pypdf/blob/e7e4ffc7e74fd3f1a191bc63527bee0d7986be1f/pyproject.toml#L13